x-pack/platform/plugins/shared/fleet/server/services/output.ts

```diff
@@ -1142,7 +1142,10 @@ class OutputService {
     const outputSO = await this.encryptedSoClient.update<Nullable<OutputSOAttributes>>(
       SAVED_OBJECT_TYPE,
       outputIdToUuid(id),
-      updateData
+      updateData,
+      {
+        mergeAttributes: false,
```

Comment:
Our fleet API route for updating outputs uses a `PUT` HTTP verb, but our update of the saved object was performing partial updates and merging existing fields with fields included in the update. This had the effect of persisting the `secrets` field when it was omitted in the update. This change makes sure this is a full replace of the object.

Reply:
Fleet APIs are not following strictly REST and all of our PUT APIs are in reality PATCH, like most of the Kibana APIs

Reply:
Gotcha! I'll refactor towards the PATCH behavior instead. Looks like anyways this change broke a bunch of tests which I missed.